io/io/src/TFileMerger.cxx

```diff
@@ -386,7 +388,7 @@ Bool_t TFileMerger::MergeRecursive(TDirectory *target, TList *sourcelist, Int_t
    // Get the dir name
    TString path(target->GetPath());
    // coverity[unchecked_value] 'target' is from a file so GetPath always returns path starting with filename:
-   path.Remove(0, path.Last(':') + 2);
+   path.Remove(0, std::strlen(target->GetFile()->GetPath()));
```

Comment:
I can not see how this is similar to the old code (which was removing the prefix/protocol) ....

Could you add some test (in io/io/test) that checks that this is functioning as expected?

Reply:
> I can not see how this is similar to the old code

```
user@host $ root https://root.cern.ch/js/files/hsimple.root
root [0] 
Attaching file https://root.cern.ch/js/files/hsimple.root as _file0...
(TFile *) 0x55a59e7cce90
root [1] target = _file0->mkdir("dir")
(TDirectory *) @0x7ffd3594db48
root [2] TString path
(TString &) ""[0]
root [3] path = target->GetPath()
(TString &) "https://root.cern.ch/js/files/hsimple.root:/dir"[47]
root [4] path.Remove(0, path.Last(':') + 2)
(TString &) "dir"[3]
root [5] path = target->GetPath()
(TString &) "https://root.cern.ch/js/files/hsimple.root:/dir"[47]
root [6] path.Remove(0, std::strlen(target->GetFile()->GetPath()))
(TString &) "dir"[3]
root [7] target->GetFile()->GetPath()
(const char *) "https://root.cern.ch/js/files/hsimple.root:/"
```